contracts/staking/contracts/src/vaults/StakingPoolRewardVault.sol

```diff
@@ -172,35 +173,76 @@ contract StakingPoolRewardVault is
     /// Note that this is only callable by the staking contract, and when
     /// not in catastrophic failure mode.
     /// @param poolId Unique Id of pool.
-    /// @param poolOperatorShare Fraction of rewards given to the pool operator, in ppm.
-    function registerStakingPool(bytes32 poolId, uint32 poolOperatorShare)
+    /// @param operatorShare Fraction of rewards given to the pool operator, in ppm.
+    function registerStakingPool(
+        bytes32 poolId,
+        address payable operatorAddress,
+        uint32 operatorShare
+    )
         external
         onlyStakingContract
         onlyNotInCatastrophicFailure
     {
         // operator share must be a valid fraction
-        if (poolOperatorShare > PPM_DENOMINATOR) {
-            LibRichErrors.rrevert(LibStakingRichErrors.InvalidPoolOperatorShareError(
+        if (operatorShare > PPM_DENOMINATOR) {
+            LibRichErrors.rrevert(LibStakingRichErrors.OperatorShareError(
+                LibStakingRichErrors.OperatorShareErrorCodes.OPERATOR_SHARE_MUST_BE_BETWEEN_0_AND_100,
                 poolId,
-                poolOperatorShare
+                operatorShare
             ));
         }
 
         // pool must not exist
-        Balance memory balance = balanceByPoolId[poolId];
-        if (balance.initialized) {
+        Pool memory pool = poolById[poolId];
+        if (pool.initialized) {
```

Comment:
Slipped me during the last review: Just like below, I actually think that it would be more efficient to update the `poolById[poolId]` storage pointer directly since only one read is being performed and `SLOAD` operations are fairly expensive now. I'm going to add this to my weekly Todos to investigate more thoroughly, but I think that this would be a good change in the meantime.

Reply:
Nice! I like the update to a storage pointer! +1 for readability